python/pyspark/pandas/frame.py

```diff
@@ -3989,9 +3989,14 @@ def insert(
             )
 
         if is_name_like_tuple(column):
-            if len(column) != len(self.columns.levels):
-                # To be consistent with pandas
-                raise ValueError('"column" must have length equal to number of column levels.')
+            if isinstance(self.columns, pd.MultiIndex):
+                if len(column) != len(self.columns.levels):
+                    # To be consistent with pandas
+                    raise ValueError('"column" must have length equal to number of column levels.')
+            else:
+                raise NotImplementedError(
+                    "Tuple-like name is not supported to non-MultiIndex column"
```

Comment:
What does this mean?

Reply:
I meant tuple-like name (e.g. `("a", "b")`) is not supported for DataFrame with `Index` unlike pandas.

**pandas**
```python
>>> pdf[("a", "b")] = [4, 5, 6]
>>> pdf
   a  (a, b)
0  1       4
1  2       5
2  3       6
```

**pandas API on Spark**
```python
>>> psdf[("a", "b")] = [4, 5, 6]
Traceback (most recent call last):
...
KeyError: 'Key length (2) exceeds index depth (1)'
```

Reply:
Do you think should we make it work in this PR ?

Reply:
Let me address to make it work in this PR!